File: Scala Kompics/src/se/sics/kompics/scala/performancetest/kompics/nnar/NNAR.scala
package se.sics.kompics.scala.performancetest.kompics.nnar

import se.sics.kompics.PortType
import se.sics.kompics.Event

// Read from register
case class NNARRead extends Event

// Write value to register
case class NNARWrite(value: Any) extends Event

// Return read value from register
case class NNARReadReturn(value: Any) extends Event

// Confirm write to register
case class NNARWriteReturn extends Event

object NNAR extends PortType {
	PortType.preloadInstance(this);
	request(classOf[NNARRead]);
	request(classOf[NNARWrite]);
	indication(classOf[NNARReadReturn]);
	indication(classOf[NNARWriteReturn]);
}

File: Java Kompics/src/se/kth/ii2202/experiments/nnar/NNAR.java
package se.kth.ii2202.experiments.nnar;

import se.sics.kompics.PortType;

/**
 *
 * @author carbone
 */
public class NNAR extends PortType{
    {
        request(NNARread.class);
        request(NNARwrite.class);
        indication(NNARreadreturn.class);
        indication(NNARwritereturn.class);
    }
}
